cncserver.js

```diff
@@ -1125,6 +1126,18 @@ function serialPortReadyCallback() {
       return;
     }
 
+    // Setting the value of the power to the pen
+    if (typeof inPen.power !== "undefined") {
+      powers = botConf.get('penpower');

```

Comment:
In JS if you just set the value of a random variable, it's instantiated in the global scope, and this is a big smelly no-no! Make sure and put the word `var` behind it to initialize it as a variable in the given function scope. Netbeans should color global JS vars very differently, should be a dead givaway.